src/pages/ApproveAction/GenericApprovalScreen.tsx

```diff
@@ -69,34 +81,179 @@ export function GenericApprovalScreen() {
     action,
     network,
   });
+  const {
+    isGaslessOn,
+    setIsGaslessEligible,
+    getGaslessEligibility,
+    isGaslessEligible,
+    gaslessFundTx,
+    setIsGaslessOn,
+    isFundProcessReady,
+    fundTxHex,
+    fundTxDoNotRertyError,
+    setGaslessDefaultValues,
+    createGaslessOffscreen,
+    closeGaslessOffscreen,
+    solutionHex,
+    isGaslessFundStarted,
+    setIsGaslessFundStarted,
+  } = useNetworkFeeContext();
+
+  const { captureEncrypted } = useAnalyticsContext();
+  const [gaslessError, setGaslessError] = useState('');
 
-  const { displayData, context } = action ?? {};
+  const { displayData, context, signingData } = action ?? {};
   const hasFeeSelector = action?.displayData.networkFeeSelector;
   const isFeeValid =
+    (isGaslessOn && isGaslessEligible) ||
     !hasFeeSelector ||
     (!feeError && !isCalculatingFee && hasEnoughForNetworkFee);
 
   useEffect(() => {
     if (!displayData?.network?.chainId) {
       return;
     }
+    const getGaslessStatus = async (
+      chainId: number | string,
+      fromAddress?: AddressLike | null,
+      nonce?: number | null,
+    ) => {
+      const gaslessEligibility = await getGaslessEligibility(
+        chainId,
+        fromAddress?.toString(),
+        nonce,
+      );
+      setIsGaslessEligible(gaslessEligibility);
+    };
 
     setNetwork(getNetwork(displayData.network.chainId));
-  }, [getNetwork, displayData?.network?.chainId]);
+
+    if (!signingData || signingData.type !== RpcMethod.ETH_SEND_TRANSACTION) {
+      setIsGaslessEligible(false);
+      return;
+    }
+    getGaslessStatus(
+      displayData.network.chainId,
+      signingData?.data?.from,
+      signingData?.data?.nonce,
+    );
+  }, [
+    displayData,
+    displayData?.network?.chainId,
+    getGaslessEligibility,
+    getNetwork,
+    isGaslessEligible,
+    setIsGaslessEligible,
+    setIsGaslessOn,
+    signingData,
+  ]);
+
+  useEffect(() => {
+    createGaslessOffscreen();
```

Comment:
Why does the frontend need to control this? Shouldn't this be fully automatic from the backend? What happens if one approval screen opens an offscreen and the other approval window closes it immediately?